core/schema/container.go

```diff
@@ -715,8 +718,64 @@ type containerFromArgs struct {
 	Address string
 }
 
-func (s *containerSchema) from(ctx context.Context, parent *core.Container, args containerFromArgs) (*core.Container, error) {
-	return parent.From(ctx, args.Address)
+func (s *containerSchema) from(ctx context.Context, parent dagql.Instance[*core.Container], args containerFromArgs) (inst dagql.Instance[*core.Container], _ error) {
+	bk, err := parent.Self.Query.Buildkit(ctx)
+	if err != nil {
+		return inst, fmt.Errorf("failed to get buildkit client: %w", err)
+	}
+	platform := parent.Self.Platform
+
+	refName, err := reference.ParseNormalizedNamed(args.Address)
+	if err != nil {
+		return inst, fmt.Errorf("failed to parse image address %s: %w", args.Address, err)
+	}
+	// add a default :latest if no tag or digest, otherwise this is a no-op
+	refName = reference.TagNameOnly(refName)
+
+	if refName, isCanonical := refName.(reference.Canonical); isCanonical {
+		ctr, err := parent.Self.FromCanonical(ctx, refName)
+		if err != nil {
+			return inst, err
+		}
+
+		return dagql.Instance[*core.Container]{
+			Constructor: dagql.CurrentID(ctx),
+			Self:        ctr,
+			Class:       parent.Class,
+			Module:      parent.Module,
+		}, nil
+	}
```

Comment:
The updates here were another situation where we want a `NodeFunc` so that we can *conditionally* return a modified ID depending on the input.

In the past I had resigned to adding "internal" apis prefixed with `__` (to hide them from codegen) to do this, but realized the approach used here of:
1. creating a `dagql.Instance` and setting the constructor to `dagql.CurrentID` when you *don't* need any modifications 
2. calling the current field with a `Select` and modified args when you *do* need some modifications
 
is much much cleaner; avoided the need for `__internalFrom` and such (which pollutes telemetry besides being ugly).

cc @rajatjindal @helderco @vito, related to recent conversations on discord.

Reply:
Neat :tada: we could do this with quite a few things :smile: 

Should we have a helper for this `Instance` incantation?

Reply:
Yeah good idea, adding `dagql.NewInstanceForCurrentID` (mouthful, but it's at least "what it says on the tin" and that's what autocomplete is for anyways). Generalized it slightly so it can handle parent and self being different types too.